boot/bootutil/src/loader.c

```diff
@@ -715,10 +743,10 @@ boot_validate_slot(int slot, struct boot_status *bs)
     }
 
     hdr = boot_img_hdr(&boot_data, slot);
-    if (boot_magic_is_erased(flash_area_erased_val(fap), hdr->ih_magic) ||
-            hdr->ih_flags & IMAGE_F_NON_BOOTABLE) {
+    if (boot_check_header_erased(slot) == 0 || (hdr->ih_flags & IMAGE_F_NON_BOOTABLE)) {
```

Comment:
There is no distinction between an error message and a "header not erased".
Currently, it looks like `boot_check_header_erased` has a "tristate" return value, `0` for `true`, `-1` for `false` and `BOOL_EFLASH` for error.
Would it be better to only return the error state, and have  "is erased" set through an output argument?
This way the error code could be checked separately in a better way than the current return scheme.

Reply:
This is a good point... I decided to change it to `-1` here because if makes no difference. What we want is a bool behaviour. Returning `BOOT_EFLASH` always meant some flash driver error, which many times is not tested, but still has a semantic meaning if there comes a day where something can be done about it.